### DISCUSSION TITLE
What is the recommended approach for exposing APIs like GraphQL?

### DISCUSSION THREAD

**Discussion by j-sauer** [eclipse-edc/Connector#1578] [Q&A] [answered]
Hi,
I have a GraphQL that I want to expose via a data space connector.
What is the recommended approach (e.g. extensions) for doing this currently?

**Comment by jimmarino** [comment, marked as answer]
You can build an extension that exposes whatever API you want. Take a look at the onboarding videos and documentation for details.

**Reply by j-sauer** [reply]
So there is currently no extensions suitable for exposing a GraphQL API but one should develop a new one, correct?

**Reply by jimmarino** [reply]
EDC supports multiple APIs such as elements of IDS multipart and a RESTful API for various management operations. Users can create their own custom APIs using technologies such as GraphQL where it makes sense. For example, GraphQL could be used to offer an API on cached contract offers stored by the Federated Catalog Crawler.

**Reply by florianrusch-zf** [reply]
@j-sauer do you want that the EDC exposes an GraphQL API? So something like @jimmarino mentioned in his comment, that you can fetch the contract offers via GraphQL? Or do you want to expose an existing GraphQL API **through** the EDC? So you have a Backend-Data-Service with an GraphQL API and you want to access this from outside through the EDC?

**Reply by j-sauer** [reply]
The latter :)

**Reply by florianrusch-zf** [reply]
I'm not very familiar with GraphQL, but there is already something like a proxy-flow mechanism. Take a look here: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/extensions/data-plane-transfer/data-plane-transfer-sync

**Reply by j-sauer** [reply]
thanks that's what I was searching for
